pkg/fx/storage/provider.go

```diff
@@ -0,0 +1,111 @@
+package storage
+
+import (
+	"context"
+
+	"github.com/storacha/go-ucanto/client"
+	"github.com/storacha/go-ucanto/principal"
+	"go.uber.org/fx"
+
+	"github.com/storacha/piri/pkg/config/app"
+	"github.com/storacha/piri/pkg/pdp"
+	"github.com/storacha/piri/pkg/service/blobs"
+	"github.com/storacha/piri/pkg/service/claims"
+	"github.com/storacha/piri/pkg/service/replicator"
+	"github.com/storacha/piri/pkg/service/storage"
+	"github.com/storacha/piri/pkg/service/storage/ucan"
+	"github.com/storacha/piri/pkg/store/receiptstore"
+)
+
+var Module = fx.Module("storage",
+	fx.Provide(
+		fx.Annotate(
+			NewStorageService,
+			fx.As(new(storage.Service)),
+			fx.As(new(ucan.BlobAllocateService)),
+			fx.As(new(ucan.BlobAcceptService)),
+			fx.As(new(ucan.PDPInfoService)),
+			fx.As(new(ucan.ReplicaAllocateService)),
+		),
+	),
+)
+
+// StorageServiceParams contains all dependencies for the storage service
+type StorageServiceParams struct {
+	fx.In
+
+	Config       app.AppConfig
+	ID           principal.Signer
+	Blobs        *blobs.BlobService
+	Claims       *claims.ClaimService
+	PDP          pdp.PDP `optional:"true"`
+	ReceiptStore receiptstore.ReceiptStore
+	Replicator   *replicator.Service
+}
+
+// storageServiceWrapper wraps the storage service to implement the storage.Service interface
+type storageServiceWrapper struct {
+	id           principal.Signer
+	blobs        *blobs.BlobService
+	claims       *claims.ClaimService
+	pdp          pdp.PDP
+	receiptStore receiptstore.ReceiptStore
+	replicator   *replicator.Service
+	uploadConn   client.Connection
+}
+
+// NewStorageService creates a new storage service
+func NewStorageService(params StorageServiceParams, lc fx.Lifecycle) (storage.Service, error) {
```

Comment:
alternatively, `NewService` would be consistent with the naming in other packages